libs/community/langchain_community/document_loaders/recursive_url_loader.py

```diff
@@ -231,19 +237,26 @@ async def _async_get_child_links_recursive(
         )
         async with self._lock:  # type: ignore
             visited.add(url)
-        try:
-            async with session.get(url) as response:
-                text = await response.text()
-                if self.check_response_status and 400 <= response.status <= 599:
-                    raise ValueError(f"Received HTTP status {response.status}")
-        except (aiohttp.client_exceptions.InvalidURL, Exception) as e:
-            logger.warning(
-                f"Unable to load {url}. Received error {e} of type "
-                f"{e.__class__.__name__}"
-            )
-            if close_session:
-                await session.close()
-            return []
+
+        async with semaphore:
+            try:
+                async with session.get(url) as response:
+                    text = await response.text()
+                    if self.check_response_status and 400 <= response.status <= 599:
+                        raise ValueError(f"Received HTTP status {response.status}")
+            except (aiohttp.client_exceptions.InvalidURL, Exception) as e:
+                logger.warning(
+                    f"Unable to load {url}. Received error {e} of type "
+                    f"{e.__class__.__name__}"
+                )
+                if close_session:
+                    await session.close()
+                return []
+            
+            # Wait if the concurrent requests limit is reached
+            if semaphore.locked():
```

Comment:
what's this needed for, doesn't L241 already handle this?

Reply:
I included it to allow the loader to complete as many requests as possible once the limit is reached before sending other requests. My idea was to insert a delay to avoid HTTP errors caused by an excessive number of simultaneous requests.

If you think that's redundant I can quickly update the code.

Reply:
@baskaryan could you please tell me if I need to update the code?